providers/fcontext.rb

```diff
@@ -0,0 +1,50 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  escaped_file_spec = Regexp.escape(new_resource.file_spec)
+  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
```

Comment:
I have little exp too :(
Let's keep it like that and add `#XXX: Might run on unrelated files` above
it.
BTW, I think that you can get the noregex by escaping and then splitting
before the first '\' (which would indicate a regex), and adding a `*` at
the end.

*Nitzan Raz*
http://backslasher.net

On Tue, Feb 24, 2015 at 3:57 PM, Jörg Herzinger <notifications@github.com>
wrote:

> In providers/fcontext.rb
> <https://github.com/BackSlasher/chef-selinuxpolicy/pull/3#discussion_r25252976>
> :
>
> > @@ -0,0 +1,50 @@
> > +# Support whyrun
> > +def whyrun_supported?
> > +  true
> > +end
> > +
> > +use_inline_resources
> > +
> > +# Create if doesn't exist, do not touch if fcontext is already registered
> > +action :add do
> > +  escaped_file_spec = Regexp.escape(new_resource.file_spec)
> > +  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
>
> Hmm, I get what you mean, but restorecon just applies what is set in
> selinux. I don't have too much experience with selinux, but do you thing it
> is that evil?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BackSlasher/chef-selinuxpolicy/pull/3/files#r25252976>
> .
>